### PR TITLE
Support DecimalType in `sort` and `take` kernels

### LINKED ISSUE #1137: support sort for decimal data type/ decimalarray
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

In the https://github.com/apache/arrow-datafusion/issues/1522, we need to sort results by decimal column.

The core of the sorting decimal array is based on the sort kernel.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

Closes #1137 

# Rationale for this change
 
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by liukun4515** [comment]
We can fill follow-up prs to optimize decimal sort.

**Comment by liukun4515** [comment]
@alamb @houqp PTAL

File: arrow/src/compute/kernels/take.rs [hunk 1]
```diff
@@ -483,6 +487,38 @@ where
...
+/// `take` implementation for decimal arrays
+fn take_decimal128<IndexType>(
+    decimal_values: &DecimalArray,
+    indices: &PrimitiveArray<IndexType>,
+) -> Result<DecimalArray>
+where
+    IndexType: ArrowNumericType,
+    IndexType::Native: ToPrimitive,
+{
+    // TODO optimize decimal take and construct decimal array from MutableBuffer
```
Comment (alamb) [inline arrow/src/compute/kernels/take.rs:499]: This would be a great follow on task / PR for someone -- I'll file a ticket

**Review by alamb — APPROVED** [review]
Thanks @liukun4515  looks good to me

File: arrow/src/compute/kernels/take.rs (hunk 1 above)
Comment (alamb) [reply arrow/src/compute/kernels/take.rs:499]: https://github.com/apache/arrow-rs/issues/1190

Comment (liukun4515) [reply arrow/src/compute/kernels/take.rs:499]: thanks

**Pull request merged**